### PR TITLE
added CONTRIBUTING.md - testing and codestyle guidelines

### PR DESCRIPTION
**Pull request by eahefnawy**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
-TEST_SERVERLESS_EMAIL=yada@google.com  
-TEST_SERVERLESS_REGION1=us-east-1  
-TEST_SERVERLESS_REGION2=us-west-2  
-TEST_SERVERLESS_STAGE1=development  
-TEST_SERVERLESS_STAGE2=production  
-TEST_SERVERLESS_LAMBDA_ROLE=ENTERARNHERE  
-TEST_SERVERLESS_AWS_ACCESS_KEY=  
-TEST_SERVERLESS_AWS_SECRET_KEY=  
+### Testing Guidelines
+We're crazy about testing! we testing everything and we aim for 100% test coverage. For test coverage, we use Istanbul locally, and coveralls on our main repo.
```
Comment (flomotlik) [inline CONTRIBUTING.md]: we testing everything -> We test everything

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+## Pull Requests
+Before you open a PR, please make sure you open an issue for it to discuss your changes first. This is just to ensure that your changes are within our roadmap and that your PR will be accepted and your efforts won't be wasted.
 
-s-test-prj
+If your PR solves a specific open bug on our issue tracker or a feature request that we've decided to support, in that case opening a new issue is unnecessary. We just want to make sure we're aware of any work you'll be doing in advance so that we can help you along the way. Regardless of what you do, we appreciate you for just being in our community.
 
-1. Copy these Environment Variables in your Serverless tests/.env file to use the Serverless Test AWS Account and Test Project. Don’t change their values:
+Once you start working on a PR, please follow our Testing and Code Style guidelines bellow. We can't accept your PR if it fails in any of these two critical areas. Whenever you make or update a PR, 3 checks fire on this PR from Travis-CI and Coveralls. If the builds fail, we can't accept your PR.
```
Comment (pmuens) [inline CONTRIBUTING.md]: bellow --> below

File: CONTRIBUTING.md (hunk 1 above)
Comment (eahefnawy) [inline CONTRIBUTING.md]: LOL

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,28 +1,47 @@
...
+
+When you open an issue, we'll add relevant labels to that issue asap to reflect category, area of code, status and priority of the issue. We'll keep those labels updated to keep you updated on our progress. 
+
+If your issue is a bug report, please make sure you include an accurate/detailed description of the issue you're facing and how to reproduce it. Make sure you mention your Serverless version, as well as your environment (node/npm version, OS...etc)
+
+## Providing Support
+This simplest thing you can do to help us move forward and make an impact on our progress is to simply provide support to other people having some difficulties with their Serverless project. You can do that by replying to issues on Github, or simply chatting with other community members on Gitter. Your very presence is enough to make us happy.
+
+## Improving Documentation
+Maintaining and updating the docs on a regular basis is a hard task. The more eyeballs on the docs the higher quality it'll get and the less chances there will be for typos and confusion. We keep our docs in our `serverless-docs` repo. If you see any issues with our docs, simply open an issue or a PR.
```
Comment (flomotlik) [inline CONTRIBUTING.md]: We keep documentation in the docs folder, not a separate repo. After that change GTM

**Pull request merged**